### PR TITLE
feat: create [SOLICITOR] role (FPLA-907)

### PR DESCRIPTION
**Pull request by toireasam**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: ccd-definition/CaseEvent/Gatekeeping.json [hunk 1]
```diff
@@ -209,5 +209,33 @@
...
+    "ShowEventNotes": "N",
+    "EndButtonLabel": "Save and continue"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "ID": "addStatementOfServiceGatekeeping",
+    "Name": "Add statement of service (c9)",
+    "Description": "Add statement of service",
+    "DisplayOrder": 13,
```
Comment (akomisarek) [inline ccd-definition/CaseEvent/Gatekeeping.json]: I think display order with those two new elements is a bit off.

File: ccd-definition/CaseEvent/PREPARE_FOR_HEARING.json [hunk 2]
```diff
@@ -200,5 +200,33 @@
...
+    "ShowEventNotes": "N",
+    "EndButtonLabel": "Save and continue"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "ID": "addStatementOfService-PREPARE_FOR_HEARING",
+    "Name": "Add statement of service (c9)",
+    "Description": "Add statement of service",
+    "DisplayOrder": 13,
```
Comment (akomisarek) [inline ccd-definition/CaseEvent/PREPARE_FOR_HEARING.json]: I think display order with those two new elements is a bit off.

File: ccd-definition/CaseEvent/Submitted.json [hunk 3]
```diff
@@ -219,5 +219,19 @@
...
     "ShowEventNotes": "N",
     "EndButtonLabel": "Save and continue"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "CARE_SUPERVISION_EPO",
+    "ID": "limitedUploadDocuments",
+    "Name": "Documents",
+    "Description": "Upload documents",
+    "DisplayOrder": 1,
```
Comment (akomisarek) [inline ccd-definition/CaseEvent/Submitted.json]: If this is one, should be placed next to 1 with display order. But probably should be 15 to mimic the doc placement?

Comment (toireasam) [reply ccd-definition/CaseEvent/Submitted.json]: done

File: ccd-definition/CaseEvent/Gatekeeping.json (hunk 1 above)
Comment (toireasam) [reply ccd-definition/CaseEvent/Gatekeeping.json]: done

File: ccd-definition/CaseEvent/PREPARE_FOR_HEARING.json (hunk 2 above)
Comment (toireasam) [reply ccd-definition/CaseEvent/PREPARE_FOR_HEARING.json]: done

File: ccd-definition/UserProfile.json [hunk 4]
```diff
@@ -62,6 +62,55 @@
     "WorkBasketDefaultCaseType": "CARE_SUPERVISION_EPO",
     "WorkBasketDefaultState": "Open"
   },
+  {
+    "LiveFrom": "01/01/2017",
+    "UserIDAMId": "raghu@croydon.gov.uk",
+    "WorkBasketDefaultJurisdiction": "PUBLICLAW",
+    "WorkBasketDefaultCaseType": "CARE_SUPERVISION_EPO",
+    "WorkBasketDefaultState": "Open"
+  },
```
Comment (akomisarek) [inline ccd-definition/UserProfile.json]: Do we have those users locally? If not please remove - that file is not used in pipeline, it's confusing I know sorry!

File: service/src/main/resources/application.yaml [hunk 5]
```diff
@@ -138,15 +138,15 @@ appinsights:
 
 fpl:
   local_authority_email_to_code:
-    mapping: 'fpla.local=>FPLA;swansea.gov.uk=>SA;hillingdon.gov.uk=>HN;swindon.gov.uk=>SN;wiltshire.gov.uk=>SNW'
+    mapping: 'fpla.local=>FPLA;swansea.gov.uk=>SA;hillingdon.gov.uk=>HN;swindon.gov.uk=>SN;wiltshire.gov.uk=>SNW;croydon.gov.uk=>CR'
```
Comment (akomisarek) [inline service/src/main/resources/application.yaml]: I haven't seen new users added as part of this pr in add-users, do we need those changes?

File: ccd-definition/UserProfile.json (hunk 4 above)
Comment (toireasam) [reply ccd-definition/UserProfile.json]: Removed :) I may have added this when we seen the user profile error in the pipeline before i knew it was casued by ccd

File: service/src/main/resources/application.yaml (hunk 5 above)
Comment (toireasam) [reply service/src/main/resources/application.yaml]: removed :) i think i was adding this here to test why the notification error was happening for raghu!

**Pull request merged**